cmd/influxd/config.go

```diff
@@ -251,6 +246,38 @@ func ParseConfig(s string) (*Config, error) {
 	return c, nil
 }
 
+type Graphite struct {
+	Address       string `toml:"address"`
+	Database      string `toml:"database"`
+	Enabled       bool   `toml:"enabled"`
+	Port          uint16 `toml:"port"`
+	Protocol      string `toml:"protocol"`
+	NamePosition  string `toml:"name-position"`
+	NameSeparator string `toml:"name-separator"`
+}
+
+// Default carbon port per http://graphite.readthedocs.org/en/1.0/feeding-carbon.html
+const defaultGrahitePort = 2004
+
+// ConnnectionString returns the connection string for this Graphite config in the form
+// host:port.
+func (g Graphite) ConnectionString(defaultBindAddr string) string {
```

Comment:
I don't know if it makes a copy or not, but I do know that in general there is typically no performance gain by using a pointer of not in Go.  While there are exceptions, this tends to be the rule.  So, the debate is typically for readability.  Only use pointer receivers if you intend to mutate, then the signature itself lets people know your intent.  Again, the community tends to be divided.